Modules/vc-module-notifications/VirtoCommerce.NotificationsModule.Data/Services/NotificationMessageService.cs

```diff
@@ -19,19 +18,25 @@ public class NotificationMessageService : INotificationMessageService
     {
         private readonly Func<INotificationRepository> _repositoryFactory;
         private readonly IEventPublisher _eventPublisher;
+        private readonly INotificationService _notificationService;
 
-        public NotificationMessageService(Func<INotificationRepository> repositoryFactory, IEventPublisher eventPublisher)
+        public NotificationMessageService(Func<INotificationRepository> repositoryFactory, IEventPublisher eventPublisher, INotificationService notificationService)
         {
             _repositoryFactory = repositoryFactory;
             _eventPublisher = eventPublisher;
+            _notificationService = notificationService;
         }
 
         public async Task<NotificationMessage[]> GetNotificationsMessageByIds(string[] ids)
         {
             using (var repository = _repositoryFactory())
             {
                 var messages = await repository.GetMessagesByIdsAsync(ids);
-                return messages.Select(n => n.ToModel(AbstractTypeFactory<NotificationMessage>.TryCreateInstance())).ToArray();
+                return messages.Select(n =>
```

Comment:
Why we can load related  notifications on the repository level as navigation properties with further converting into domain types in entity models?